app/services/info_request_batch_zip.rb

```diff
@@ -0,0 +1,98 @@
+##
+# This service streams a ZIP file with all incoming/ outgoing messages and
```

Comment:
I'm not sure I'd call this a "service". To me, a service should be more like a [command method](https://www.martinfowler.com/bliki/CommandQuerySeparation.html) – i.e. it does some work and returns `nil` (or maybe `true` or `false` as an indicator of success).

This feels like a legitimate object – it _is a_ Zip. Personally I'd make this `InfoRequestBatch::Zip`.

We don't really have a good precedent on this though, so this is more of a suggestion/comment – happy to keep it as it is for now.